mldsa/src/sign.c

```diff
@@ -629,7 +634,8 @@ int crypto_sign_signature(uint8_t sig[CRYPTO_BYTES], size_t *siglen,
     /* To be on the safe-side, make sure *siglen has a well-defined */
     /* value, even in the case of error.                            */
     *siglen = 0;
-    return -1;
+    result = -1;
+    goto cleanup;
```

Comment:
This can be argued to be unnecessary, since no buffer has been written to.

Reply:
A compiler could move 
```
  pre[0] = 0;
  pre[1] = (uint8_t)ctxlen;
``` 
to before conditional? 
Anyway, I did not want to get into reasoning if steps can be reordered by the compiler or not.